src/kademlia/routing_table.cpp

```diff
@@ -1125,8 +1139,7 @@ void routing_table::add_router_node(udp::endpoint router)
 // pinged == false)
 void routing_table::heard_about(node_id const& id, udp::endpoint const& ep)
 {
-	// only when the node_id pass the verification, add it to routing table.
-	if (m_settings.enforce_node_id && !verify_id(id, ep.address())) return;
+	if (!verify_node_address(m_settings, id, ep.address())) return;
```

Comment:
there's a place pretty early on after receiving a packet (in the subscription callback from the dht socket) where we check to see if the packet looks like a DHT packet (basically, does it start with "d" and end with "e" and is it large enough to contain the mandatory keys). I think at this point it would also make sense to check if it's from an IPv4 address. I guess that check would have to be removed once steven's patch is in though